benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/step/RangeWithStep.scala

```diff
@@ -0,0 +1,56 @@
+package cloud.benchflow.dsl.definition.configuration.goal.explorationspace.step
+
+import cloud.benchflow.dsl.definition.errorhandling.BenchFlowDeserializationException
+
+import scala.annotation.tailrec
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * @author Jesper Findahl (jesper.findahl@gmail.com)
+ *         created on 2017-06-23
+ */
+object RangeWithStep {
+
+  def stepList[T](
+    range: List[T],
+    step: T,
+    func: (T, T) => T)(implicit num: Numeric[T]): Try[List[T]] = {
+
+    val list = generateList(List(range.head), step, range(1), func)
+
+    if (list.size <= 2) {
```

Comment:
I check that the range has two elements in the list, that the step function is valid and that the step itself is an integer. Not sure what we can test beyond that before running. If you have any ideas that would be helpful.